README.md

```diff
@@ -66,8 +66,16 @@ When you have implemented the tool, answer the questions below, commit it to Git
 
 How does your method for extracting features work?
 
-What is the complexity of the algorithm, as a function of the size of the two input files? When you answer this, you need to know that you can get the list of chromosomse from a `query.Table` in constant time, but it does, of course, take longer to run through all the lines in it.
+We reformat our bed file, such that the four different lines in the file are distinguished. We do the same for our query file, such that we can loop through the query file and bed file. When looping through we match chromosomes from query to the bed table, such that we can extract the chromosome and all of its three other contents to our output file.
+
+What is the complexity of the algorithm, as a function of the size of the two input files? When you answer this, you need to know that you can get the list of chromosomes from a `query.Table` in constant time, but it does, of course, take longer to run through all the lines in it.
+
+We first parse the bed file into the table, thus running through bed file n times, hereafter we parse query m times, and then we run through each line of the query m times to detect overlaps. Thus resulting in O(n+m^2)
```

Comment:
Why would you run through the bed file $n$ times to fill the table? The code looks like this, and it certainly looks like you only look at each line once:

```python
    for line in args.bed:
         tabel.add_line(parse_line(line))
```

Since adding to the table runs in $O(1)$, and so does `parse_line()`, this would be $O(n)$.

If you really *did* run through the bed file $n$ times, then your complexity would have an $O(n^2)$ component, but it doesn't, so either the time or the description of why is wrong. It is the description. Parsing the file takes $O(n)$.

You also don't run through each line in the query file $m$ times.

```python
    for line in args.query:
         chrom, start, end = line.split()
         ...
```

It only costs you $O(m)$ to run through all the query files, not $O(m^2)$.

For each query line, however, you potentially have to look at all the bed lines (if they are all on the same chromosome and you are querying that chromosome). So, in the worst case, it will take $O(n)$ to handle one query or $O(nm)$ to handle $m$ queries.